From 3523d49e8e573ded7ae9609369dc6df31d6bfbc9 Mon Sep 17 00:00:00 2001
From: uweigand <uweigand@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 28 Jul 2014 14:32:13 +0000
Subject: [PATCH] 	PR libobjc/61920
 	* encoding.c (rs6000_special_adjust_field_align_p): Use definition
 	that matches the 4.9 branch ABI.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_9-branch@213127 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libobjc/encoding.c |    3 ++-
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/libobjc/encoding.c b/libobjc/encoding.c
index a603b8c..c476f22 100644
--- a/libobjc/encoding.c
+++ b/libobjc/encoding.c
@@ -192,7 +192,8 @@ _darwin_rs6000_special_round_type_align (const char *struc, int comp, int spec)
    ? MAX (MAX (COMPUTED, SPECIFIED), 64)				\
    : MAX (COMPUTED, SPECIFIED));})
 
-#define rs6000_special_adjust_field_align_p(FIELD, COMPUTED) 0
+#define rs6000_special_adjust_field_align_p(FIELD, COMPUTED) \
+ (TARGET_ALTIVEC && TREE_CODE (TREE_TYPE (FIELD)) == VECTOR_TYPE)
 
 /* Skip a variable name, enclosed in quotes (").  */
 static inline
-- 
1.7.1

